starter_code/src/movies.js

```diff
@@ -1,15 +1,148 @@
 /* eslint no-restricted-globals: 'off' */
 
 // Iteration 1: Ordering by year - Order by year, ascending (in growing order)
+function orderByYear(moviesArray) {
+
+    //slice without arguments creates a copy
+    let copiedMoviesArray = moviesArray.slice()
+
+    let sortedArray = copiedMoviesArray.sort(function (a, b) {
+        if (a.year === b.year) {
+            if (a.title > b.title) {
+                return 1;
+            } else {
+                return -1;
+            }
+        }
+        return a.year - b.year
+    })
+    return sortedArray
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
+function howManyMovies(steveArray) {
+    let result = 0
+    steveArray.filter(function (dramaSteve) {
+        if (dramaSteve.genre === "Drama" && dramaSteve.director === "Steven Spielberg") {
+            result++;
+        } else {
+            return 0;
+        }
+    })
```

Comment:
if here we use .length we can get the length of new Array that filter just created so we dont need result but its good way of thinking nice